PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UpdateCheckpointAsyncCheck.java

```diff
@@ -0,0 +1,201 @@
+        @Override
+        public void visitMethodCallExpression(@NotNull PsiMethodCallExpression expression) {
+            super.visitMethodCallExpression(expression);
+
+            // Check if the rule should be skipped
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+
+            if (expression.getMethodExpression() == null || expression.getMethodExpression().getReferenceName() == null) {
+                return;
+            }
+
+            // Check if the method call is updateCheckpointAsync()
+            if ("updateCheckpointAsync".equals(expression.getMethodExpression().getReferenceName())) {
+
+                // Get the method name following the updateCheckpointAsync() method call
+                String followingMethod = getFollowingMethodName(expression);
+
+                // Check if the following method is `subscribe` and
+                //  Check if the updateCheckpointAsync() method call is called on an EventBatchContext object
+                if ("subscribe".equals(followingMethod) && isCalledOnEventBatchContext(expression)) {
+                    holder.registerProblem(expression, RULE_CONFIG.getAntiPatternMessageMap().get("with_subscribe"));
+                }
+                //  Check if the updateCheckpointAsync() method call is called on an EventBatchContext object
+                else if (followingMethod == null || (!followingMethod.equals("block") && !followingMethod.equals("block_with_timeout")) && isCalledOnEventBatchContext(expression)) {
+                    holder.registerProblem(expression, RULE_CONFIG.getAntiPatternMessageMap().get("no_block"));
+                }
+            }
+        }
+
+        /**
+         * This method checks if the method call is following a updateCheckpointAsync() method call.
+         * If the method call is following a updateCheckpointAsync(), the method name is returned.
+         *
+         * @param expression The method call expression
+         * @return The method name if the method call is following a specific method call, null otherwise
+         */
+        private String getFollowingMethodName(PsiMethodCallExpression expression) {
+
+            // Get the parent of the method call expression
+            PsiElement parent = expression.getParent();
+
+            if (!(parent instanceof PsiReferenceExpression)) {
+                return null;
+            }
+            PsiElement grandParent = parent.getParent();
+
+            if (grandParent instanceof PsiMethodCallExpression) {
+
+                PsiMethodCallExpression parentCall = (PsiMethodCallExpression) grandParent;
+
+                // Get the method name from the parent call
+                String methodName = parentCall.getMethodExpression().getReferenceName();
+
+                // Check if the method name is in the list of methods to check
+                if (RULE_CONFIG.getMethodsToCheck().contains(methodName)) {
+                    if ("block".equals(methodName)) {
```

Comment:
A suggestion: when we write rules, if we see that these strings ("block", "subscribe" etc..) are used often in different rules, it's a good practice to declare them as static members of constant.java file.

Reply:
it only shows up in this rule so far so I can make block a constant instead:)